pkg/pillar/cmd/downloader/syncop.go

```diff
@@ -323,12 +323,17 @@ func sourceFailureError(ip, ifname, url string, err error) {
 
 func getDatastoreCredential(ctx *downloaderContext,
 	dst types.DatastoreConfig) (zconfig.CredentialBlock, error) {
-	cred := utils.PrepareCipherCred(dst.ApiKey, dst.Password)
-	status, cred, err := utils.GetCipherCredentials(agentName,
-		dst.CipherBlockStatus, cred)
-	if status.IsCipher && len(status.Error) == 0 {
+	var cred zconfig.CredentialBlock
+	var status types.CipherBlockStatus
+	var err error
+	status, cred, err = utils.GetCipherCredentials(agentName,
+		dst.CipherBlockStatus)
```

Comment:
Why do we even call this when IsCipher is not set?

Reply:
Agree, I will update it.